pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -521,6 +521,23 @@ func (s *DiscoveryServer) WorkloadUpdate(id string, labels map[string]string, _
 		s.WorkloadsByID[id] = &Workload{
 			Labels: labels,
 		}
+		adsClientsMutex.RLock()
+		for _, connection := range adsClients {
+			// if the workload has envoy proxy and connected to server,
+			// then do a full xDS push for this proxy;
+			// otherwise:
+			//   case 1: the workload has no sidecar proxy, no need xDS push at all.
+			//   case 2: the workload xDS connection has not been established,
+			//           also no need to trigger a full push here.
+			if connection.modelNode.IPAddresses[0] == id {
+				// There is a possibility that the pod comes up later than endpoint.
+				// So no endpoints add/update events after this, we should request
+				// full push immediately to speed up sidecar startup.
+				go s.startPush(versionInfo(), s.globalPushContext(), false, nil, id)
```

Comment:
The upstream PR seems to be triggering a full push for the connection in this case, shouldn't we do that, too?

Reply:
Going through the logs, it seems sufficient to do an incremental push. Missing clusters and listeners are pushed out immediately when a late pod event comes in, before a full push is triggered.

Reply:
My only request here is: if we stick with incremental push, we should adjust the comment above this line that still mentions a full push.

Reply:
Apparently this is a misunderstanding on my part about incremental vs full push